libethash-cuda/CUDAMiner.cpp

```diff
@@ -515,6 +513,7 @@ void CUDAMiner::search(
                     s_gridSize, s_blockSize, stream, buffer, current_nonce, m_parallelHash);
 
         }
+        updateHashRate(batch_size, s_numStreams);
```

Comment:
Why move it here ?

Reply:
To reduce the number of calls. Here it's only called once per outer loop, instead of once per the stream inner loop.

I have one more push to this PR to move the solution submit logic out of the way. As it is it delays the restarting of each stream.